db/sg_replicate_cfg.go

```diff
@@ -333,16 +333,17 @@ func (rc *ReplicationConfig) Redacted() *ReplicationConfig {
 
 // sgReplicateManager should be used for all interactions with the stored cluster definition.
 type sgReplicateManager struct {
-	cfg                        cbgt.Cfg                      // Key-value store implementation
-	loggingCtx                 context.Context               // logging context for manager operations
-	heartbeatListener          *ReplicationHeartbeatListener // node heartbeat listener for replication distribution
-	localNodeUUID              string                        // nodeUUID for this SG node
-	activeReplicators          map[string]*ActiveReplicator  // currently assigned replications
-	activeReplicatorsLock      sync.RWMutex                  // Mutex for activeReplications
-	clusterUpdateTerminator    chan struct{}                 // Terminator for cluster update retry
-	clusterSubscribeTerminator chan struct{}                 // Terminator for cluster change monitoring
-	closeWg                    sync.WaitGroup                // Teardown waitgroup for subscribe and retry goroutines
-	dbContext                  *DatabaseContext
+	cfg                               cbgt.Cfg                      // Key-value store implementation
+	loggingCtx                        context.Context               // logging context for manager operations
+	heartbeatListener                 *ReplicationHeartbeatListener // node heartbeat listener for replication distribution
+	localNodeUUID                     string                        // nodeUUID for this SG node
+	activeReplicators                 map[string]*ActiveReplicator  // currently assigned replications
+	activeReplicatorsLock             sync.RWMutex                  // Mutex for activeReplications
+	clusterUpdateTerminator           chan struct{}                 // Terminator for cluster update retry
+	clusterSubscribeTerminator        chan struct{}                 // Terminator for cluster change monitoring
+	closeWg                           sync.WaitGroup                // Teardown waitgroup for subscribe and retry goroutines
+	dbContext                         *DatabaseContext              // reference to the parent DatabaseContext
+	defaultCheckpointIntervalOverride time.Duration                 // The value to be used when a per-ActiveReplicator CheckpointInterval is not set
```

Comment:
Given that ActiveReplicatorConfig.CheckpointInterval is already test-only, and I don't see any use cases for different intervals in the same test, it seems like we could just have one manager-level value for CheckpointInterval that all tests could set directly (with whatever locking is necessary).  I think that would avoid having an 'interval' and an 'interval override'?

Reply:
I still need some way of passing down an interval through `sgReplicateManager->ActiveReplicator->Active(Push|Pull)Replicator->Checkpointer`, and the most convinent place for that is an ActiveReplicatorConfig (outside of a global variable + locking, which I'm not keen on doing)

I've moved all of the default value handling up into `sgReplicateManager` to make things a bit simpler. Anything managed (by sgReplicateManager) sets checkpoints once every 5 seconds, unless the manager has non-default value explicitly set (for tests)

Doing this meant I was able to remove some of the "disable time-based checkpointing" parts of non-managed tests.